### PR TITLE
Autoupdate manifest 2845

### LINKED ISSUE #2845: Some package with checkver without autoupdate
## Bug Report

Some package have a checkver without autoupdate that we can fix:

- apimtemplate: 0.7 (scoop version is 0.5)
- dhall: 1.40.1 (scoop version is 1.39.0)
- dnscrypt-proxy: 2.1.1 (scoop version is 2.1.0)
- fio: 3.27 (scoop version is 3.20)
- k0s: 1.22.2 (scoop version is 0.13.1)
- megatools: 1.10.3 (scoop version is 1.10.2)
- mockery: 2.9.4 (scoop version is 2.8.0)
- rainbow: 1.1.0 (scoop version is 1.0.1)
- uutils-coreutils: 0.0.8 (scoop version is 0.0.6)

### PR DESCRIPTION
**Pull request by Valinor**
fix #2845

update and add autoupdate to some manifest :

apimtemplate: 0.7 (scoop version is 0.5)
dhall: 1.40.1 (scoop version is 1.39.0)
dnscrypt-proxy: 2.1.1 (scoop version is 2.1.0)
fio: 3.27 (scoop version is 3.20)
k0s: 1.22.2 (scoop version is 0.13.1)
megatools: 1.10.3 (scoop version is 1.10.2)
mockery: 2.9.4 (scoop version is 2.8.0)
pngquant: 2.16.0 (scoop version is 2.15.1)
rainbow: 1.1.0 (scoop version is 1.0.1)
uutils-coreutils: 0.0.8 (scoop version is 0.0.6)

### CODE REVIEW

**Comment by pratikpc** [comment]
@Valinor can you test Fio once?

The run failed for Fio

**Comment by Valinor** [comment]
image

Yes, I tested it, I don't understand the problem with the failed run, which seem a stopped runner without finish the test.
By the way, seem to me Test are ok in the last iteration :  
image

**Comment by pratikpc** [comment]
No the bigger issue is that verify failed.

**Comment by pratikpc** [comment]
Trying again.

Check the Actions outputs once.

**Comment by pratikpc** [comment]
Workflow run failed for verify

File: bucket/mockery.json [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
         }
     },
     "bin": "mockery.exe",
-    "checkver": "github"
+    "checkver": "github",
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/vektra/mockery/releases/download/v$version/mockery_$version_Windows_x86_64.tar.gz"
+            }
```
Comment (pratikpc) [inline bucket/mockery.json:18]: Mockery has a checksums.txt file

Could we use that for hashes?

https://github.com/vektra/mockery/releases/download/v2.9.4/checksum.txt

**Comment by Valinor** [comment]
> Trying again.
> 
> Check the Actions outputs once.

How and where? (where is the outpout for verify?) Where are verify workflow log? Sorry for my ignorance...

**Comment by pratikpc** [comment]
Check https://github.com/ScoopInstaller/Main/actions/runs/1404368672 once

**Comment by pratikpc** [comment]
> > Trying again.
> > Check the Actions outputs once.
> 
> How and where? (where is the outpout for verify?) Where are verify workflow log? Sorry for my ignorance...

No issues on that one. There's a lot of GitHub that even I don't understand properly

Currently on Scoop a GitHub Action runs whenever verify is called. However this particular action failed so we don't get the output too.

**Comment by Valinor** [comment]
> @Valinor can you test Fio once?
> 
> The run failed for Fio

```
Manifest: /github/workspace/bucket/fio.json

INFO: Hashes

The SSL connection could not be established, see inner exception.
At /SCOOP/apps/scoop/current/lib/install.ps1:130 char:9
+         throw $e
+         ~~~~~~~~
+ CategoryInfo          : OperationStopped: (:) [], HttpRequestException
+ FullyQualifiedErrorId : The SSL connection could not be established, see inner exception.
```

It seem more like a script problem, a timeout, or anything else, and not specific to Fio.
Address in Fio manifest are all reachable, with no certificate problem.

I will split this commit to check.

**Comment by rashil2000** [comment]
This error is coming up in way too many places. 
We might need to handle TLS versions in this function - https://github.com/ScoopInstaller/Scoop/blob/35b2a42ede584ae898460d209c04ec48fda1e027/lib/install.ps1#L364

Here are some older issues related to this:
https://github.com/ScoopInstaller/Scoop/issues/3746
https://github.com/ScoopInstaller/Scoop/issues/2040
https://github.com/ScoopInstaller/Scoop/pull/2074
https://github.com/ScoopInstaller/Scoop/issues/4002

Here's some more info - https://stackoverflow.com/questions/41618766/powershell-invoke-webrequest-fails-with-ssl-tls-secure-channel

**Comment by Valinor** [comment]
@pratikpc  et @rashil2000 can you approuve the squash/merge : 
@pratikpc, it's a bug for the failed verification. I can't do anything, and the autoupdate can help for all theses manifests.

File: bucket/fio.json [hunk 2]
```diff
@@ -1,20 +1,32 @@
 {
-    "version": "3.20",
+    "version": "3.27",
     "description": "Flexible I/O Tester",
     "homepage": "https://bsdio.com/fio/",
```
Comment (pratikpc) [inline bucket/fio.json:4]: Is the homepage for Fio correct @Valinor ?

This is Rebecca Cran's fio package homepage right?

If this isn't, we could fix it in a future PR

**Review by pratikpc — APPROVED** [review]

File: bucket/apimtemplate.json [hunk 3]
```diff
@@ -1,19 +1,26 @@
 {
-    "version": "0.5",
+    "version": "0.7",
     "description": "Azure API Management DevOps Resource Kit",
     "homepage": "https://github.com/Azure/azure-api-management-devops-resource-kit",
     "license": "MIT",
     "depends": "azure-cli",
```
Comment (pratikpc) [inline bucket/apimtemplate.json:6]: Should we change it to a suggestion instead?

Future PR of course

File: bucket/rainbow.json [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
-    "url": "https://github.com/arsham/rainbow/releases/download/v1.0.1/rainbow_windows_v1.0.1.zip",
-    "hash": "bab9af652642e1e7066b8c9c174a9621e9bed2e8acd1823001bd11b922488d24",
+    "url": "https://github.com/arsham/rainbow/releases/download/v1.1.0/figurine_windows_v1.1.0.zip",
+    "hash": "b7e37fb7b61c4ba2e34b3e67960ceda9a9253ebfeda3bf84add1361ad631b98c",
     "extract_dir": "deploy",
     "bin": "rainbow.exe",
-    "checkver": "github"
+    "checkver": "github",
+    "autoupdate": {
+        "url": "https://github.com/arsham/rainbow/releases/download/v$version/figurine_windows_v$version.zip"
```
Comment (pratikpc) [inline bucket/rainbow.json:12]: Can we add hash extraction for this?

The releases page contains Binary checksums

b7e37fb7b61c4ba2e34b3e67960ceda9a9253ebfeda3bf84add1361ad631b98c figurine_windows_v1.1.0.zip

**Review by rashil2000 — APPROVED** [review]

File: bucket/fio.json (hunk 2 above)
Comment (Valinor) [reply bucket/fio.json:4]: I don't know, there is no official webpage, only the git link : https://git.kernel.dk/cgit/fio/
I think the Rebecca page is best.

File: bucket/rainbow.json (hunk 4 above)
Comment (Valinor) [reply bucket/rainbow.json:12]: Yes I will do that

File: bucket/apimtemplate.json (hunk 3 above)
Comment (Valinor) [reply bucket/apimtemplate.json:6]: Yes I will do that

Comment (pratikpc) [reply bucket/apimtemplate.json:6]: @Valinor for this should we instead create a new PR labelled good first issue?

Would help new Scoop devs who wish to participate but haven't participate in the development and if 2 weeks later no PR is open, we manually update it?

**Pull request merged**